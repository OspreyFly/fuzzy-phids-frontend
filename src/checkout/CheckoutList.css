.Checkout-list {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    gap: 20px; /* Space between cards */
    background-color: #0F0F0F;
  }
  
  .Checkout-list .card {
    flex: 1 0 21%; /* Flex basis for 4 cards per row */
    min-width: 250px; /* Minimum width of each card */
    max-width: 250px; /* Maximum width of each card */
    box-sizing: border-box; /* Include padding and border in width calculation */
  }
  
  .Checkout-list p.lead {
    text-align: center;
    font-size: 1.2em;
    margin-top: 20px;
  }

  #summary {
    color: #008170;
  }

  .warning {
    color: #008170; 
  }