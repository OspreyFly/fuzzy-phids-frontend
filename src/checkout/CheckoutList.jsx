import React from 'react';
import InsectCard from '../insects/InsectCard';
import LoadingSpinner from '../common/LoadingSpinner';
import CheckoutForm from './CheckoutForm';
import './CheckoutList.css';

const CheckoutList = ({ currentUser, getSummary, cart, shopping, setCart, addToCart, removeFromCart }) => {
    if (!cart) return <LoadingSpinner />;
    if (cart.length < 1) return <div><h1 className="warning">Your cart is empty!</h1></div>;
    return (
        <div >
            {getSummary()}
            <div className="Checkout-list">
                {cart.map(c => (
                    <InsectCard class="card"
                        id={c.id}
                        species={c.species}
                        price={c.price}
                        url_image={c.url_image}
                        cart={cart}
                        addToCart={addToCart}
                        removeFromCart={removeFromCart}
                        shopping={shopping}
                    />
                ))}
            </div>
            <div className="total-price">
            </div>
            <CheckoutForm currentUser={currentUser} cart={cart} setCart={setCart} />
        </div>
    );
}

export default CheckoutList;
