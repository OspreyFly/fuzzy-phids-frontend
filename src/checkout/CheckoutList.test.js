import React from 'react';
import { render } from '@testing-library/react';
import CheckoutList from './CheckoutList';

jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
}));

const cart = [{id:1, species: "a", price: 1}, {id:2, species: "b", price: 1}];
const currentUser = {id: 1, username: "testuser"};

describe('CheckoutList', () => {
let getSummary = () => {
  return [];
}
  test('renders correctly', () => {
    const {asFragment} = render(<CheckoutList currentUser={currentUser}cart={cart} getSummary={getSummary}/>);
    
    expect(asFragment()).toMatchSnapshot();
  });
});