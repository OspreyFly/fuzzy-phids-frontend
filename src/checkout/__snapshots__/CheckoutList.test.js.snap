// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutList renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Checkout-list"
    >
      <div
        class="InsectCard card"
      >
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            a
          </h6>
          <h3>
            $1
          </h3>
          <button
            class="animated-minus-button"
          >
            -
          </button>
        </div>
      </div>
      <div
        class="InsectCard card"
      >
        <div
          class="card-body"
        >
          <h6
            class="card-title"
          >
            b
          </h6>
          <h3>
            $1
          </h3>
          <button
            class="animated-minus-button"
          >
            -
          </button>
        </div>
      </div>
    </div>
    <div
      class="total-price"
    />
    <form
      class="checkout-form"
    >
      <div>
        <label
          for="phone"
        >
          Phone:
        </label>
        <input
          class="checkout-form"
          id="phone"
          maxlength="30"
          minlength="1"
          name="phone"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="delivery_address"
        >
          Delivery Address:
        </label>
        <input
          id="delivery_address"
          maxlength="10"
          minlength="1"
          name="delivery_address"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
